Model/Service/Sync.php

```diff
@@ -0,0 +1,261 @@
+            $op = new UpsertProduct($account, $this->nostoHelperUrl->getActiveDomain($store));
+            $op->setResponseTimeout(self::RESPONSE_TIMEOUT);
+            foreach ($collectionBatch as $productIndex) {
+                if (!$productIndex->getInSync()) {
+                    $op->addProduct($productIndex->getNostoProduct());
+                }
+            }
+            try {
+                $op->upsert();
+                $this->tickBenchmark(self::BENCHMARK_SYNC_NAME);
+            } catch (\Exception $upsertException) {
+                $this->getLogger()->exception($upsertException);
+            } finally {
+                // We will set this as in sync even if there was failures
+                $collectionBatch->markAsInSyncCurrentItemsByStore($store);
+            }
+        });
+        $this->logBenchmarkSummary(self::BENCHMARK_SYNC_NAME, $store);
+        try {
+            $totalDeleted = $this->purgeDeletedProducts($store);
+            $this->getLogger()->info(
+                sprintf(
+                    'Removed total of %d products from index for store %s',
+                    $totalDeleted,
+                    $store->getCode()
+                )
+            );
+        } catch (NostoException $e) {
+            $this->getLogger()->exception($e);
+        }
+    }
+
+    /**
+     * Defines product index as in sync
+     *
+     * @param ProductIndexInterface $productIndex
+     * @throws \Exception
+     * @return void
+     */
+    public function markAsInSync(ProductIndexInterface $productIndex)
+    {
+        if (!$productIndex->getInSync()) {
+            $productIndex->setInSync(true);
+            $this->indexRepository->save($productIndex);
+        }
+    }
+
+    /**
+     * @param int[] $productIds
+     * @param Store $store
+     * @return void
+     */
+    public function markAsInSyncByProductIdsAndStoreId(array $productIds, Store $store)
+    {
+        try {
+            $this->nostoIndexCollectionFactory->create()->markAsInSync($productIds, $store);
+        } catch (\Exception $e) {
+            $this->getLogger()->exception($e);
+        }
+    }
+
+    /**
+     * Discontinues products in Nosto and removes indexed products from Nosto product index
+     *
+     * @param NostoIndexCollection $collection
+     * @param Store $store
+     * @return int number of deleted products
+     * @throws NostoException
+     */
+    public function deleteIndexedProducts(NostoIndexCollection $collection, Store $store)
+    {
+        if ($collection->getSize() === 0) {
+            return 0;
+        }
+        $account = $this->nostoHelperAccount->findAccount($store);
+        if ($account instanceof NostoSignupAccount === false) {
+            throw new NostoException(sprintf('Store view %s does not have Nosto installed', $store->getName()));
+        }
+        $this->startBenchmark(self::BENCHMARK_DELETE_NAME, self::BENCHMARK_DELETE_BREAKPOINT);
+        $totalDeleted = 0;
```

Comment:
Why are we counting this? Either we count both the upserts and deletes or we remove them both. I say we just remove it.

Reply:
We can ditch this